src/backend/functions/orchestrator/lib/index.js

```diff
@@ -0,0 +1,86 @@
+const AWS = require('aws-sdk');
+const R = require('ramda');
+const converter = AWS.DynamoDB.Converter;
+
+// snakeToCamel :: String -> String
+const snakeToCamel = R.replace(/([_][a-z])/g, R.compose(R.replace('_', ''), R.toUpper));
+
+function noop() {}
+
+// convertEnvVars :: {k: v} -> {k: v}
+const convertEnvVars = R.pipe(
+    R.toPairs,
+    R.map(R.adjust(0, R.compose(snakeToCamel, R.toLower))),
+    R.fromPairs,
+    R.evolve({subnets: subnets => [...subnets.split(',').map(R.trim)]})
+);
+
+// startTranscription :: AWS.ECS -> AWS.DDB -> Promise {k: v}
+const startTranscription = R.curry((ecs, ddb, options) => {
+   const params = {
+      taskDefinition: options.taskName,
+      cluster: options.cluster,
+      launchType: 'FARGATE',
+      networkConfiguration: {
+         awsvpcConfiguration: {
+            subnets: options.subnets,
+            assignPublicIp: 'DISABLED'
+         }
+      },
+      overrides: {
+         containerOverrides: [
+            {
+               name: 'transcriber',
+               environment: [
+                  {
+                     name: 'MEDIA_URL',
+                     value: options.mediaUrl
+                  }
+               ]
+            }
+         ]
+      }
+   };
+
+   return ecs.runTask(params)
+       .promise()
+       .then(({tasks}) => {
+         return ddb.put({
```

Comment:
maybe just me, kind of feels like this doing 2 things - starting transcription + recording transcription. recording the transcription will probably be used by other states too so might be worth separating?

Reply:
Yeah, I 'd thought something similar but I want to implement the other states first to see whether an abstraction emerges naturally from that. This is the currently simplest implementation and will definitely not be the final version.